dash_bio/utils/gene_expression_reader.py

```diff
@@ -1,57 +1,151 @@
+import GEOparse as gp
 import pandas as pd
 import tempfile
 
 
 # pylint: disable=unnecessary-lambda
 
-def parse_tsv(
+def read_soft_file(
         contents='',
         filepath='',
```

Comment:
This looks weird to me. It looks like we use this function by specifying either `contents` or `filepath`, is that correct? Then, I understand why the other argument doesn't matter and comes with a meaningless default value... But I don't find it very clean. Let's take sides and favour either `contents` or `filepath`.

Reply:
So if we favour contents, for example, would we simply ignore `contents` if the filepath is not `None`?

Reply:
Let's do this in a subsequent PR. There are actually a few parsers that use this pattern -- e.g., https://github.com/plotly/dash-bio/blob/96d43e90c4598b42282ee27b28e827c8e0f6ec7e/dash_bio/utils/protein_reader.py#L29 
I'd prefer to get those all done in one go, if that's OK!

Reply:
If we favour contents, `contents=''` would become `contents` (has to be specified) and `filename=''` could be fine (since it's meaningless, it might as well be `filename=None`). Alternatively, if standard, conventional names exist (but I guess we would mean basename rather than filename), we can use them as default values. I'm not familiar with the format, but I guess along the lines of `contents='data', filepath='data.soft'`.

Okay to change it in a dedicated PR.

Reply:
Great -- please open an issue for that in the repo!

Reply:
Done #330 -- resolving then.